pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -226,9 +339,109 @@ func (s *GameServerSelector) Validate(field string) ([]metav1.StatusCause, bool)
 		}
 	}
 
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) && (s.Counters != nil || s.Lists != nil) {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: "Feature CountsAndLists must be enabled if Counters or Lists are specified",
+			Field:   field,
+		})
+	}
+
+	if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		if s.Counters != nil {
+			causes = append(causes, s.validateCounters(field)...)
+		}
+		if s.Lists != nil {
+			causes = append(causes, s.validateLists(field)...)
+		}
+	}
+
 	return causes, len(causes) == 0
 }
 
+// validateCounters validates that the selection field has valid values for CounterSelectors
```

Comment:
That's a lot of tests 😄

Reply:
Same tests as for player tracking, just a bunch more fields.